packages/editor/src/components/post-actions/index.js

```diff
@@ -22,17 +22,39 @@ const { Menu, kebabCase } = unlock( componentsPrivateApis );
 
 export default function PostActions( { postType, postId, onActionPerformed } ) {
 	const [ isActionsMenuOpen, setIsActionsMenuOpen ] = useState( false );
-	const { item, permissions } = useSelect(
+	const { item, permissions, additionalContext } = useSelect(
```

Comment:
I just learned that, in `trunk`, we are already augmenting the data with extra info: `permissions`.

Even it's obscured by the fact that here's done via direct call to `getEntityRecordPermissions` while in the site editor is done via the `useEntityRecordsWithPermissions` utility. It was introduced [here](https://github.com/WordPress/gutenberg/pull/63923/files#diff-f6d2ea7cd82485550ff502f466f05c286fe0cb0c444921ba001394a06280fe55). Before that change, at registration time, we'd wrap the `isEligible` function so that we had access to the registry.

The advantage of doing that at registration time was that every consumer didn't have to care about augmenting the data. Also, we already have eligibility logic for the actions at registration time (see [private-actions](https://github.com/WordPress/gutenberg/blob/6d36c1b8ee1c3c68601a61959bc1c2602ee107d1/packages/editor/src/dataviews/store/private-actions.ts#L105)): though that logic doesn't depend on the item, while the permissions or the homepage info does.

@youknowriad could you share more of the rationale for that change? I dug a bit but I didn't find why we preferred not to do that.